part-1.py

```diff
@@ -4,17 +4,34 @@
 # the appropriate comment.
 
 # factorial
-
+def factorial(n):
+    if n==1 or n==0:
+        return 1
```

Comment:
This base case is reaching ahead.  With recursion, I recommend trying to find the simplest base cases possible:
```suggestion
    if n==0:
        return 1
```